src/plugins/intel_gpu/include/intel_gpu/primitives/rnn.hpp

```diff
@@ -0,0 +1,189 @@
+                                                                activation_func::hyperbolic_tan,
+                                                                activation_func::hyperbolic_tan},
+              const std::vector<activation_additional_params>& activation_params = {},
+              const lstm_weights_order& offset_order = lstm_weights_order::iofz,
+              const ov::op::RecurrentSequenceDirection direction = ov::op::RecurrentSequenceDirection::FORWARD,
+              const padding& output_padding = padding(),
+              const int num_outputs = 1)
+        : primitive_base<PType>(id, {x}, num_outputs, {optional_data_type()}, {output_padding}),
+        x(x),
+        initial_hidden_state(initial_hidden_state),
+        initial_cell_state(initial_cell_state),
+        W(W),
+        R(R),
+        B(B),
+        seq_lenghts(seq_lenghts),
+        clip(clip),
+        input_forget(input_forget),
+        activations(activations),
+        activation_params(activation_params),
+        offset_order(offset_order),
+        direction(direction) {
+        std::vector<std::string> pids{initial_hidden_state.pid, initial_cell_state.pid, W.pid, R.pid, B.pid, seq_lenghts.pid};
+        for (auto pid : pids) {
+            if (!pid.empty()) {
+                primitive_base<PType>::input.push_back(pid);
+            }
+        }
+    }
+
+    input_info x;
+    input_info initial_hidden_state;
+    input_info initial_cell_state;
+    input_info W;
+    input_info R;
+    input_info B;
+    input_info seq_lenghts;
+    /// @brief Cell clip threshold T. It is applied to the input of activations [-T, T]. No clip is applied if it is not specified.
+    float clip;
+    bool input_forget;
+    /// @brief A list of 3 activation functions for the input, output, forget, cell, and hidden.
+    std::vector<activation_func> activations;
+    /// @brief Optional scaling values used by some activation functions. The values are consumed in the order of activation functions.
+    std::vector<activation_additional_params> activation_params;
+    /// @brief Weights, recurrent weights, and biases order. [iofz] : ONNX, [ifoz] : Caffe
+    lstm_weights_order offset_order;
+    /// @brief direction of LSTMSequence - only FORWARD or REVERSE, currently BIDIRECTIONAL not supported
+    ov::op::RecurrentSequenceDirection direction;
+
+    int num_directions() const {
+        return direction == ov::op::RecurrentSequenceDirection::BIDIRECTIONAL ? 2 : 1;
+    }
+
+    size_t hash() const override {
+        size_t seed = primitive::hash();
+        seed = hash_combine(seed, x.pid);
+        seed = hash_combine(seed, initial_hidden_state.pid);
+        seed = hash_combine(seed, initial_cell_state.pid);
+        seed = hash_combine(seed, seq_lenghts.pid);
+        seed = hash_combine(seed, W.pid);
+        seed = hash_combine(seed, R.pid);
+        seed = hash_combine(seed, B.pid);
```

Comment:
Comparison and hashing of the primitive ids prevents primitive reuse if we have multiple instances of the same op. So you shall just hash/compare only presence flag to all inputs. As an example you can use `convolution` op.

Reply:
done